Update documentation for Getting Started with data migration

jugal-chauhan opened the pull request:
### Description
This change updates the "Getting started with data migration" documentation to improve clarity and accuracy. Specifically, it:
- Adds a new prerequisite about verifying the CDKToolkit Stack status.
- Updates the instructions for managing cluster passwords in AWS Secrets Manager.
- Clarifies the steps for setting up Bootstrap instance access.

### Issues Resolved
Adds more clarification

### Version
All

### Frontend features
N/A

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Naarcha-AWS commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -20,6 +20,7 @@ Before using this quickstart, make sure you fulfill the following prerequisites:
 * Verify that your migration path [is supported]({{site.url}}{{site.baseurl}}/migration-assistant/is-migration-assistant-right-for-you/#migration-paths). Note that we test with the exact versions specified, but you should be able to migrate data on alternative minor versions as long as the major version is supported.
 * The source cluster must be deployed Amazon Simple Storage Service (Amazon S3) plugin.
 * The target cluster must be deployed.
+* Verify that the CDKToolkit Stack exists and is set to `CREATE_COMPLETE`. Refer to [this documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) on how to bootstrap your AWS account in the required AWS Region
```
```suggestion
* Verify that the `CDKToolkit` stack exists and is set to `CREATE_COMPLETE`. For more information about  on how to bootstrap your AWS account in the required AWS Region, see [the CDKTooklit documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html).
```

jugal-chauhan commented:
Hello Team, I initially forgot to add my signoff on my first commit. I noticed another commit made by @Naarcha-AWS, which still had a spacing issue. I addressed that issue in a 3rd commit with correct signoff. FInally, I have squashed all 3 commits into a single one that contains my signoff. Please do review or get back to me for additional details. Thanks!

Naarcha-AWS commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -70,6 +71,7 @@ Use the following steps to set up Bootstrap instance access:
     {% include copy.html %}
 
 3. Name the policy, for example, `SSM-OSMigrationBootstrapAccess`, and then create the policy by selecting **Create policy**.
+4. Attach this newly created policy to your EC2 instance's IAM Role.
```
```suggestion
4. Attach the newly created policy to your EC2 instance's IAM Role.
```

Naarcha-AWS commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -107,14 +109,15 @@ To use these steps, make sure you fulfill the following prerequisites:
+Use the following steps to configure and deploy RFS, deploy Migration Assistant and verify installation of required stacks:
 
-1. Add the target cluster password to AWS Secrets Manager as an unstructured string. Be sure to copy the secret Amazon Resource Name (ARN) for use during deployment.
+1. Add the source and target cluster password as separate Secrets to AWS Secrets Manager as an unstructured string. Be sure to copy the secret Amazon Resource Name (ARN) for use during deployment.
```
```suggestion
1. Add the source and target cluster password as separate **Secrets** in the [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) as an unstructured string. Be sure to copy the secret Amazon Resource Name (ARN) for use during deployment.
```

Naarcha-AWS commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -152,54 +155,52 @@ Use the following steps to configure and deploy RFS:
     {% include copy.html %}
 
-4. Deploy the stacks:
+4. Deploy Migration Assistant and required stacks when `cdk.context.json` is fully configured:
```
```suggestion
4. Deploy Migration Assistant and any required stacks when `cdk.context.json` is fully configured:
```

peternied reviewed (commented):
Minor comments, thanks for making this update

Naarcha-AWS commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -152,54 +155,52 @@ Use the following steps to configure and deploy RFS:
-If you're creating a snapshot using migration tooling, these parameters are automatically configured. If you're using an existing snapshot, modify the `reindexFromSnapshotExtraArgs` setting with the following values:
+    
+These commands deploy the following stacks:
+* Migration Assistant network stack
+* `Reindex-from-snapshot` stack
+* Migration console stack
```
I'll fix this one.

peternied commented:
@jugal-chauhan Could you review the changes since you last reviewed and provide feedback on the unresolved comments / push updates?

jugal-chauhan reviewed (commented):
These changes look good to me. Thanks a lot @Naarcha-AWS for providing more supported documentation in a consistent writing style.

jugal-chauhan commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -152,54 +155,52 @@ Use the following steps to configure and deploy RFS:
     {% include copy.html %}
 
-4. Deploy the stacks:
+4. Deploy Migration Assistant and any required stacks when `cdk.context.json` is fully configured:
```
Agreed to the substep 4 causing confusion. My suggestion here would be to change the phrasing in such a manner : 

substep 2 : "From the same shell as the Bootstrap instance, modify the cdk.context.json file located in the /opensearch-migrations/deployment/cdk/opensearch-service-migration directory. Configure all necessary parameters in this file."

substep 3 : "Bootstrap the account and deploy the required stacks:"

substep 4 : "Deploy Migration Assistant:"

this will make sure that any user/customer completes setting up/configuring their `cdk.context.json` file in substep 2 itself. Substep 3 and Substep 4 assumes that this cdk.context.json file has already been configured in substep 2.

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -20,6 +20,7 @@ Before using this quickstart, make sure you fulfill the following prerequisites:
 * Verify that your migration path [is supported]({{site.url}}{{site.baseurl}}/migration-assistant/is-migration-assistant-right-for-you/#migration-paths). Note that we test with the exact versions specified, but you should be able to migrate data on alternative minor versions as long as the major version is supported.
 * The source cluster must be deployed Amazon Simple Storage Service (Amazon S3) plugin.
 * The target cluster must be deployed.
+* Verify that the `CDKToolkit` stack exists and is set to `CREATE_COMPLETE`. For more information about on how to bootstrap your AWS account in the required AWS Region, see [the CDKTooklit documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html).
```
```suggestion
* Verify that the `CDKToolkit` stack exists and is set to `CREATE_COMPLETE`. For more information about how to bootstrap your AWS account in the required AWS Region, see [the CDKToolkit documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html).
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -70,6 +71,7 @@ Use the following steps to set up Bootstrap instance access:
     {% include copy.html %}
 
 3. Name the policy, for example, `SSM-OSMigrationBootstrapAccess`, and then create the policy by selecting **Create policy**.
+4. Attach the newly created policy to your EC2 instance's IAM Role.
```
```suggestion
4. Attach the newly created policy to your EC2 instance's IAM role.
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -107,14 +109,23 @@ To use these steps, make sure you fulfill the following prerequisites:
 ## Step 4: Configure and deploy RFS (~20 minutes)
 
-Use the following steps to configure and deploy RFS:
+To deploy migration assistant with RFS, the following stacks must be deployed:
```
```suggestion
To deploy Migration Assistant with RFS, the following stacks must be deployed:
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -107,14 +109,23 @@ To use these steps, make sure you fulfill the following prerequisites:
+* `RFS` stack
+* `Migration console` stack
+
+Use the following steps to configure and deploy RFS, deploy Migration Assistant and verify installation of required stacks:
```
```suggestion
Use the following steps to configure and deploy RFS, deploy Migration Assistant, and verify installation of the required stacks:
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -107,14 +109,23 @@ To use these steps, make sure you fulfill the following prerequisites:
+
+Use the following steps to configure and deploy RFS, deploy Migration Assistant and verify installation of required stacks:
+
+1. Add the source and target cluster password as separate **Secrets** in the [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) as an unstructured string. Be sure to copy the secret Amazon Resource Name (ARN) for use during deployment.
```
```suggestion
1. Add the source and target cluster password as separate **Secrets** in [AWS Secrets Manager](https://docs.aws.amazon.com/secretsmanager/latest/userguide/intro.html) as an unstructured string. Be sure to copy the secret Amazon Resource Name (ARN) for use during deployment.
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
```diff
@@ -145,61 +156,55 @@ Use the following steps to configure and deploy RFS:
-
-If you're creating a snapshot using migration tooling, these parameters are automatically configured. If you're using an existing snapshot, modify the `reindexFromSnapshotExtraArgs` setting with the following values:
+    
+5. Verify from the same bootstrap instance shell that all CloudFormation stacks were installed successfully:
```
```suggestion
5. From the same Bootstrap instance shell, verify that all CloudFormation stacks were installed successfully:
```

natebower commented on _migration-assistant/deploying-migration-assistant/getting-started-data-migration.md:
````diff
@@ -145,61 +156,55 @@ Use the following steps to configure and deploy RFS:
     ```
+    {% include copy.html %}
+    
+You should receive a similar output for your region:
````
```suggestion
You should receive a similar output for your Region:
```

natebower reviewed (commented):
@Naarcha-AWS @jugal-chauhan Please see my changes and let me know if you have any questions. Thanks!
